sdk/swap-service/src/implementation/SwapManager.ts

```diff
@@ -0,0 +1,74 @@
+import { ChainInfo, ChainId } from '@summerfi/sdk-common/chains'
+import { Address, Percentage, TokenAmount } from '@summerfi/sdk-common/common'
+import { Maybe } from '@summerfi/sdk-common/utils'
+import { ISwapProvider, ISwapManager, SwapData, SwapProviderType } from '~swap-service/interfaces'
+
+export type SwapManagerProviderConfig = {
+  provider: ISwapProvider
+  chainIds: ChainId[]
+}
+
+export class SwapManager implements ISwapManager {
+  private _providersByChainId: Map<ChainId, ISwapProvider[]>
+  private _providersByType: Map<SwapProviderType, ISwapProvider>
+
+  constructor(providersConfig: SwapManagerProviderConfig[]) {
+    this._providersByChainId = new Map()
+    this._providersByType = new Map()
+
+    for (const config of providersConfig) {
+      this._registerProvider(config.provider, config.chainIds)
+    }
+  }
+
+  public async getSwapData(params: {
+    chainInfo: ChainInfo
+    fromAmount: TokenAmount
+    toMinimumAmount: TokenAmount
+    recipient: Address
+    slippage: Percentage
+    forceUseProvider?: SwapProviderType
```

Comment:
We cannot provide toMinimum as a consumer because we do not know the market price of tokens. It was the responsibility of the swap to provide us. As a consumer we just decide on the slippage.

Reply:
I see what you mean. I'll remove it then and leave only the slippage